inst/tests/tests.Rraw

```diff
@@ -16726,6 +16726,10 @@ DT = data.table(
            s4class(x=2L, y="yes", z=1)))
 test(2130.03, print(DT), output=c("   x             y", "1: 1 <ex_class[3]>",     "2: 2 <ex_class[3]>"))
 
+# .SD from grouping should be unlocked, part of #4159
+x = data.table(a=1:3, b=4:6)
+test(2131, lapply(x[ , list(dt = list(.SD)), by = a]$dt, attr, '.data.table.locked'),
```

Comment:
I think we handle that already in `split.data.table` so this PR could probably simplifies logic there https://github.com/Rdatatable/data.table/blob/d9b7ad5bd583662585032c87afb6ec6aaa5a2b21/R/data.table.R#L2310-L2314

Reply:
I think the use case here is _slightly_ different in general -- I had in mind something like returning `list(list(list(list(.SD))))`. A bit contrived, but I think it's conceivable users can return `.SD` accidentally in less straightforward ways.